CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,497 @@
+
+### Properties
+
+You can obtain more info about properties by placing the cursor in a property and pressing Ctrl+Q within Android Studio.
+
+
+#### changesetComment
+`override val changesetComment = "Add whether defibrillator is inside building"`
+message used as a changeset comment
+
+#### wikiLink
+`override val wikiLink = "Key:indoor"`
+points to the OSM Wiki page most relevant to the given quest, typically it is an added key. In this case, it is a page about [indoor=* tagging](https://wiki.openstreetmap.org/wiki/Key:indoor).
+
+#### icon
+`override val icon = R.drawable.ic_quest_defibrillator` 
+icon drawable, you can initially use any icon. 
+
+Do not worry, submissions with placeholder icons are also welcomed! In many cases icon itself was not made by the PR author.
+
+More info about icon handling [will be given later](#adding-quest-icon).
+
+#### questTypeAchievements
+`override val questTypeAchievements = listOf(LIFESAVER)`
+
+In quest achievements, list what is relevant to the given quest, see the full list of available ones in [AchievementsModule.kt](app/src/main/java/de/westnordost/streetcomplete/data/user/achievements/AchievementsModule.kt)
+
+#### getTitle
+`override fun getTitle(tags: Map<String, String>) = R.string.quest_is_defibrillator_inside_title`
+
+It is a message displayed to user, code here passes a [reference](https://developer.android.com/guide/topics/resources/string-resource) to the string. You can change it to the new, not yet existing one and use a built in tool to place text.
+
+Actual strings sit in [app/src/main/res/values/strings.xml](app/src/main/res/values/strings.xml)
+
+There are separate files with translated text, but do not worry about it - [translations are handled separately](CONTRIBUTING.md#translating-the-app).
+
+#### Form
+
+`override fun createForm() = YesNoQuestAnswerFragment()`
+
+Form defines interface used by mappers.
+
+In this case, the simplest possible is used.
+
+But sometimes more complex ones are needed, see for example [AddBridgeStructure.kt](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/bridge_structure/AddBridgeStructure.kt) 
+
+`override fun createForm() = AddBridgeStructureForm()`
+
+
+With form defined in [AddBridgeStructureForm](app/src/main/java/de/westnordost/streetcomplete/quests/bridge_structure/AddBridgeStructureForm.kt)
+### applyAnswerTo
+```
+    override fun applyAnswerTo(answer: Boolean, tags: Tags, timestampEdited: Long) {
+        tags["indoor"] = answer.toYesNo()
+    }
+``` 
+
+This code is responsible for modifying `tags` object (passed by reference).
+
+In this case it would set `indoor` tag to have either `yes` or `no` answer, depending on selection in the quest interface.
+
+Actions may include (examples from various quests):
+
+- `tags["indoor"] = answer.toYesNo()`
+- `tags.remove("amenity")` - remove key if it is present
+- `tags.updateWithCheckDate("lit", answer.toYesNo())`
```

Comment:
it probably should be mentioned that `updateWithCheckDate()` should (must?) be used if quest supports resurvey (e.g. if it has `!lit or lit older today -16 years` or similar)